README.md

```diff
@@ -61,64 +61,121 @@ that instead. There are a few external dependencies (s2n and libcrypto, for inst
 ```
 
 #### Detailed config:
-NOTE: Any key can be prefixed with a ```!``` to overwrite the config value, rather than to add to it.
+NOTE: Any key can be prefixed with a ```!``` to overwrite the config value, rather than to add to it. 
+      Any key can be prefixed with a ```+``` to force a value to be added to an array.
 
-```json
+See builder/data.py for more info/defaults/possible values.
+
+```jsonc
 {
     "name": "my-project",
     // Whether or not this project should be built
     "enabled": true,
+    // Whether or not this project needs a C/C++ compiler
+    "needs_compiler": true,
+
+    // Variables for use in interpolation throughout the config. Note that these can be overriden
+    // per host/os/target/architecture/compiler/version (see below)
+    // The following variables are pre-defined by builder internally:
+    // * host - the host OS
+    // * compiler - the compiler that will be used if project needs a compiler
+    // * version - the compiler version
+    // * target - the target OS class (linux/windows/macos/android)
+    // * arch - the target arch
+    // * cwd - the current working directory (affected by --build-dir)
+    // * source_dir - The source directory for the project being built
+    // * build_dir - The directory where intermediate build artifacts will be generated 
```

Comment:
if this is the root, don't include maybe? unless you feel its self explanatory

Reply:
I added some clarifying defaults, does that help?